### PR TITLE
DCOS-40781: Provide empty message for catalog search

### PR DESCRIPTION
**Pull request by natmegs**
Provide a message indicating when no results were found when searching Catalog page.

Closes DCOS-40781

## Testing

Go to Catalog page and try searching using strings that will match as well as strings that won't. Ensure regular filtering happens as expected and ensure non-matching strings show the empty message. Ensure 'view all' clears the filter as expected.

## Trade-offs

Seems a little convoluted to handle the search results from within `getCommunityPackagesGrid`. Maybe this could eventually be refactored into its own function.

## Screenshots

Before: 
before

After:
after

### CODE REVIEW

File: src/js/pages/catalog/PackagesTab.js [hunk 1]
```diff
@@ -208,16 +213,27 @@ class PackagesTab extends mixin(StoreMixin) {
...
-      const packagesWord = StringUtil.pluralize("service", foundPackagesLength);
+      if (foundPackagesLength < 1) {
+        const noResults = `No results were found for your search: "${
+          this.state.searchString
+        }" `;
+
+        return (
+          <div className="clearfix">
+            {noResults}
+            (<a onClick={this.clearInput}>view all</a>)
```
Comment (DanielMSchmidt) [inline src/js/pages/catalog/PackagesTab.js]: This should be `cursor: pointer`

File: src/js/pages/catalog/PackagesTab.js [hunk 2]
```diff
@@ -208,16 +213,27 @@ class PackagesTab extends mixin(StoreMixin) {
...
     const titleClasses = classNames("flush-top", {
       short: !isSearchActive,
       tall: isSearchActive
     });
 
     if (isSearchActive) {
       const foundPackagesLength = packages.getItems().length;
-      const packagesWord = StringUtil.pluralize("service", foundPackagesLength);
+      if (foundPackagesLength < 1) {
+        const noResults = `No results were found for your search: "${
```
Comment (DanielMSchmidt) [inline src/js/pages/catalog/PackagesTab.js]: I personally see no benefit in this being not inlined, I am just curious why you decided to pull this out in a variable :)

File: src/js/pages/catalog/PackagesTab.js (hunk 1 above)
Comment (Poltergeist) [reply src/js/pages/catalog/PackagesTab.js]: You may use `className="clickable"` for convenience

File: src/js/pages/catalog/PackagesTab.js (hunk 2 above)
Comment (natmegs) [reply src/js/pages/catalog/PackagesTab.js]: That was an existing variable - it is used as well [here](https://github.com/dcos/dcos-ui/blob/00d40ed81619b90ddbaaae9b6abadec5bb91a78a/src/js/pages/catalog/PackagesTab.js#L236). Should I inline in both places?

Comment (DanielMSchmidt) [reply src/js/pages/catalog/PackagesTab.js]: no, I was just wondering :D

**Comment by natmegs** [comment]
Good call, @tnguyen0. I've added an integration test (can look at last commit to isolate it from prettier fixes).

**Pull request merged**